figma-shop/src/app/pages/product-details/product-details.component.ts

```diff
@@ -10,21 +9,35 @@ import { ProductInterface } from '../../shared/interfaces/product.interface';
 })
 
 export class ProductDetailsComponent implements OnInit {
-  private product: ProductInterface;
+  public product: ProductInterface;
+  private currentCategoryProducts: ProductInterface[];
+  public currentCategoryProductsViewed: ProductInterface[];
+  private currentCategoryIndex = 0;
   public starCount = [1, 2, 3, 4, 5];
 
   constructor(
-    private actRoute: ActivatedRoute,
-    private productService: ProductService) { }
+    private actRoute: ActivatedRoute) { }
 
   ngOnInit(): void {
+    this.getProduct();
+    this.currentCategoryProducts = this.filterCurrentCategory(this.product.category);
+    this.currentCategoryProductsViewed = this.getFourProducts();
+  }
+
+  private getProduct(): void {
     this.actRoute.data.subscribe((data) => {
-      this.product = data.product;
+      this.product = data.product[0];
+      this.currentCategoryProducts = data.product[1];
     });
   }
 
-  private getProduct(): void {
-    const id = +this.actRoute.snapshot.params.id;
-    this.productService.getOneProduct(id).then(data => this.product = data);
+  private filterCurrentCategory(category: string): ProductInterface[] {
+    return this.currentCategoryProducts.filter((product: ProductInterface) => {
+      return product.category === category;
+    });
+  }
+
+  private getFourProducts(): ProductInterface[] {
```

Comment:
`getFourProducts -> getProductsByCounter(counter: number)`

Reply:
got it